lib/solidus_graphql_api/services/user/sign_in.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module SolidusGraphqlApi
+  module Services
+    module User
+      class SignIn
+        attr_reader :user
+
+        def call(email:, password:)
+          @user = Spree.user_class.find_by(email: email)
```

Comment:
Shouldn't we use [`find_for_authentication`](https://rubydoc.info/github/heartcombo/devise/master/Devise%2FModels%2FAuthenticatable%2FClassMethods:find_for_authentication)? It includes record authentication find logic like for example skips not confirmed users etc.

I think eventually we'll want to have a `GraphQLAuthenticatable` module to be included in the user class and a `find_for_graphql_authentication`, so to match Devise authentication methods logic (`find_for_authentication`, `find_for_database_authentication`, ...) but at least we should use this method to me

Reply:
👍 Thanks for the advice, I was searching for this method without success. About the `GraphQLAuthenticatable` I think we should try to touch the Solidus base classes the least possible, we have the service object for it, consider also that we'll should add a condition to include this mutation only if the project has `Devise` because without it the end developer must implement a custom `signIn` implementation.